flow/Platform.actor.cpp

```diff
@@ -3229,6 +3258,68 @@ bool isHwCrcSupported() {
 #endif
 }
 
+TmpFile::TmpFile() : filename("") {
+#ifdef __unixish__
+	createFile(defaultDir.data(), defaultPattern.data());
```

Comment:
`createFile` was implemented in dark times, when C++ had very limited support for the kind of things it's doing. I would suggest you rewrite this class and use `std::filesystem`.

Reply:
I remember trying out std::filesystem::file_size to get file size and I guess build (gcc I think) failed as std::filesystem references weren't resolved. For instance:

fdbserver/CMakeFiles/fdbserver.dir/RESTKmsConnector.actor.g.cpp.o: In function `std::filesystem::path::path<std::string, std::filesystem::path>(std::string const&, std::filesystem::path::format)':
/opt/rh/devtoolset-8/root/usr/include/c++/8/bits/fs_path.h:184: undefined reference to `std::filesystem::path::_M_split_cmpts()'

Is it OK to update make files may be to allow compilation OR there was a reason not to allow it so far?:)

Reply:
I think with `gcc` we have to link against `stdc++fs`. Adding this to cmake is fine (as long as you verify that it links it statically, but I think it should)